File: backend/backend/Areas/Main/Models/JobNotes.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Areas.Main.Models;

public class JobNotes
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    public int JobId { get; set; }
    public Job Job { get; set; }
    public int NoteId { get; set; }
    public Note Note { get; set; }
}

File: frontend/lib/models/job_notes.dart
import 'package:frontend/models/job.dart';
import 'package:frontend/models/note.dart';

class JobNotes {
  int? id;
  Job? job;
  Note? note;
  int? jobId;
  int? noteId;

  JobNotes({
    this.id,
    this.job,
    this.note,
    this.jobId,
    this.noteId
  });

  factory JobNotes.fromJson(Map<String, dynamic> json) {
    return JobNotes(
      id: json['id'],
      jobId: json['jobId'],
      job: json['job'] != null ? Job.fromJson(json['job']) : null,
      noteId: json['noteId'],
      note: json['note'] != null ? Note.fromJson(json['note']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'job': job,
      'note': note,
      'noteId': noteId,
      'jobId': jobId
    };
  }
}